root/report/components/ArtistAnnotationList.js

```diff
@@ -22,43 +26,53 @@ type Props = {
 const ArtistAnnotationList = ({
   items,
   pager,
-}: Props): React.Element<typeof PaginatedResults> => (
-  <PaginatedResults pager={pager}>
-    <table className="tbl">
-      <thead>
-        <tr>
-          <th>{l('Artist')}</th>
-          <th>{l('Type')}</th>
-          <th>{l('Annotation')}</th>
-          <th style={{width: '10em'}}>{l('Last edited')}</th>
-        </tr>
-      </thead>
-      <tbody>
-        {items.map((item, index) => (
-          <tr className={loopParity(index)} key={item.artist_id}>
-            {item.artist ? (
-              <>
-                <td>
-                  <EntityLink entity={item.artist} />
-                </td>
-                <td>
-                  {nonEmpty(item.artist.typeName)
-                    ? lp_attributes(item.artist.typeName, 'artist_type')
-                    : l('Unknown')}
-                </td>
-              </>
-            ) : (
-              <td colSpan="2">
-                {l('This artist no longer exists.')}
-              </td>
-            )}
-            <td dangerouslySetInnerHTML={{__html: item.text}} />
-            <td>{item.created}</td>
-          </tr>
-        ))}
-      </tbody>
-    </table>
-  </PaginatedResults>
-);
+}: Props): React.Element<typeof PaginatedResults> => {
+  const existingArtistItems = items.filter(item => item.artist != null);
+  const columns = React.useMemo(
+    () => {
+      const nameColumn = defineEntityColumn<ReportArtistAnnotationT>({
+        columnName: 'artist',
+        getEntity: result => result.artist ?? null,
```

Comment:
Flow doesn't know I'm filtering out results with null artist, so it needs this.

Reply:
You should be able to use `.reduce` instead of filter to work around that, though it's slightly more verbose.

Reply:
I tried that but it still gives the same Flow issues.